lib/AST/ModuleNameLookup.cpp

```diff
@@ -90,20 +95,111 @@ static bool updateOverloadSet(NamedCanTypeSet &overloads,
   return true;
 }
 
-/// After finding decls by name lookup, filter based on the given
-/// resolution kind and existing overload set and add them to \p results.
+/// Returns whether it's okay to stop recursively searching imports, given that
+/// we found something, are using a CanTypeSet to resolve overloads, and have
+/// the given \p resolutionKind.
+static bool canReturnEarly(const CanTypeSet &overloads,
```

Comment:
I'm confused about the `overloads` argument here. It doesn't seem to be used, and this method is not polymorphic. What's the reason for passing it in?
Ah--after looking at the whole file, I get it: it's statically overloading on the type of that argument.
And you juxtaposed them to make that clear.
What would probably help is a bit more "why" to go with the what. Is there a comment somewhere about why we need a Named wobulator someplaces vs a vanilla one other places?

Reply:
Yeah, I thought about this some more and this should all just be wrapped up in a strategy type.